var/spack/repos/builtin/packages/bcache/package.py

```diff
@@ -26,7 +26,7 @@ class Bcache(MakefilePackage):
 
     def setup_build_environment(self, env):
         # Add -lintl if provided by gettext, otherwise libintl is provided by the system's glibc:
-        if any("libintl." in filename.split("/")[-1] for filename in self.spec["gettext"].libs):
+        if "gettext" in self.spec and "intl" in self.spec["gettext"].libs.names:
```

Comment:
- In case it would be made conditions, it would fail in such other condition.
- When we increase the use of `mypy`: mypy would not know if is true and would issue an error about it.